crypto/evp/evp_enc.c

```diff
@@ -294,10 +294,13 @@ static int is_partially_overlapping(const void *ptr1, const void *ptr2,
 int EVP_EncryptUpdate(EVP_CIPHER_CTX *ctx, unsigned char *out, int *outl,
                       const unsigned char *in, int inl)
 {
-    int i, j, bl;
+    int i, j, bl, cmpl = inl;
+
+    if (EVP_CIPHER_CTX_test_flags(ctx, EVP_CIPH_FLAG_LENGTH_BITS))
+        cmpl /= 8;
```

Comment:
Should this actually be `cmpl = (cmpl + 7) / 8` to account for a partial byte on the end.